lib/output.js

```diff
@@ -34,3 +134,47 @@ module.exports = function(outputUri, done, fn) {
     });
   });
 };
+
+module.exports.multiOutput = function(outputs, done, fn) {
+  assert.ok(Array.isArray(outputs), "resample: targetExtent must be an array");
+
+  return tmp.dir({ unsafeCleanup: true}, function(err, workingDir) {
+    if (err) {
+      return done(err);
+    }
+
+    return fn(null, workingDir, function(err, localOutputs) {
+      if (err) {
+        return done.apply(null, arguments);
+      }
+
+      assert.ok(outputs.length === localOutputs.length,
+                util.format("Length of outputs URIs (%d) must be equal to length of local outputs (%d)", outputs.length, localOutputs.length));
+
+      return Promise
+        .resolve(_.zip(localOutputs, outputs))
+        .map(function(sourceTarget) {
+          var source = sourceTarget[0];
+          var target = sourceTarget[1];
+
+          var uri = url.parse(target);
+
+          if (uri.protocol === "s3:") {
+            return s3Upload(source, target);
+          } else {
+            var d = path.dirname(target);
+            return ensureDir(d)
+              .then(function(d) {
+                return copy(source, target);
+              });
+          }
+        }, { concurrency: os.cpus().length })
```

Comment:
No need to specify concurrency--it will already be limited by the `httpsAgent` associated with `s3Upload`.